File: DevryServiceBot/Snippets/inheritance/inherit.cpp

#include <iostream>
#include <string>
using namespace std;

class Person
{
public:
    string Name;
    string DOB;

    Person(string name, string dob)
    {
        this->Name = name;
        this->DOB = dob;
    }

    void Speak(string text)
    {
        cout << this->Name << ": " << text << endl;
    }
};

class Employee : public Person
{
public:
    int Id;

    Employee(string name, string dob, int id) : Person(name, dob)
    {
        this->Id = id;
    }

    void Quit()
    {
        this->Speak("Screw you guys I'm going home");
    }
};

int main()
{
    Person person = Person("Jim Bob", "09/01/1990");
    Employee emp = Employee("Jimmaayyyy", "01/01/2000", 12312312);

    person.Speak("Herro there");
    emp.Speak("I hate all the things");
    emp.Quit();
}

File: DevryService/Snippets/inheritance/inherit.py
class Person:
    def __init__(self, name, dob):
        self.name = name
        self.dob = dob
    
    def speak(self, text):
        print("{}: {}".format(self.name, text))
    
    def __str__(self):
        return "Name: {}\nDOB: {}".format(self.name, self.dob)
    
class Employee(Person):
    def __init__(self, name, dob, id):
        super().__init__(name, dob)
        self.id = id
    
    def quit(self):
        self.speak("Screw you guys I'm going home")
    
    def __str__(self):
        return super().__str__() + "\nId: {}".format(self.id)

person = Person("Jim Bob", "09/01/2000")
emp = Employee("Jimmayyy", "01/01/1990", 341234234)

person.speak("I'm a person")
emp.speak("I'm also a person, but also an employee")

emp.quit()

print("")
print(person)
print("")
print(emp)
# person does not have the ability to quit
#person.quit()